### PR TITLE
[Docs] Add CI-CD automation with Git

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/projects/automate-project-git-source.ipynb [hunk 1]
```diff
@@ -0,0 +1,2069 @@
+{
```
Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>If, however, your files are already compatibile for running with MLRun, you do not need to clone or extract the files - I think we can add a link that examplain about the relative hanlder</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>"# load from a private repo" - can delete this line</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>with_source_archive can be used for nuclio,serving and jobs, please add a link to the same method for nuclio and serving functions - https://docs.mlrun.org/en/stable/api/mlrun.runtimes.html?highlight=with_source_archive#mlrun.runtimes.RemoteRuntime.with_source_archive</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>link does not work</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>i think it is more accurate like this - This section describes how to set three functions that will fetch data, train the model using the fetched data, and deploy a model serving function.</p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>running a function with a relative handler is a sub option for working with remote source maybe need to be placed under the fetching the functions data and before running the function as a note </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>You can use the <code>get_function</code> method to <s>both get the function object, and</s> change the function spec - you mention in the first row that you can use is to get the object.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

File: docs/projects/git-best-practices.ipynb [hunk 2]
```diff
@@ -30,7 +30,10 @@
    "source": [
```
Comment (GiladShapira94) [inline docs/projects/git-best-practices.ipynb:30]: <p>need to fix the link - &nbsp;<a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#load-code-from-container-vs-load-code-at-runtime" target="_blank">Loading the code from container vs. loading the code at runtime</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: docs/projects/automate-project-git-source.ipynb (hunk 1 above)
Comment (GiladShapira94) [inline docs/projects/automate-project-git-source.ipynb:1]: <p>Use the code files from your project context folder on your local machine (<code>local=True</code>) can be used for debug purposes.  - maybe worth to mention that we will want to use it for debugging the code on the user local machine</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: deleted

Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: added link

Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: updated

Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: updated

File: docs/projects/git-best-practices.ipynb (hunk 2 above)
Comment (jillnogold) [reply docs/projects/git-best-practices.ipynb:30]: fixed

File: docs/projects/automate-project-git-source.ipynb (hunk 1 above)
Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: updated

Comment (GiladShapira94) [reply docs/projects/automate-project-git-source.ipynb:1]: <p>If&nbsp;<code>with_repo</code>&nbsp;is not set to&nbsp;<code>True</code>&nbsp;(default), you need to deploy the functions with&nbsp;<code>function.deploy()</code>&nbsp;to build a</p><p>container - it is relevant to to pull_at_runtime flag need to change to </p><p>If&nbsp;<code>pull_at_runtime</code>&nbsp;is not set to&nbsp;<code>True</code>&nbsp;(default), you need to deploy the functions with&nbsp;<code>function.deploy()</code>&nbsp;to build a</p><p>container.</p><p>or you can use this paragraph <code> When using with_repo=True the functions need to be deployed (function.deploy()) to build a container, unless you set project.set_source(source=source,pull_at_runtime=True) which instructs MLRun to load the git/archive repo into the function container at run time and do not require a build (this is simpler when developing, for production it’s preferred to build the image with the code)</code></p><p> </p><p> </p>

Comment (GiladShapira94) [reply docs/projects/automate-project-git-source.ipynb:1]: <p>&nbsp;<a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#running-the-function-locally" target="_blank">run jobs locally</a>  and remotely - add  remotely to the line</p><p> </p>

File: docs/projects/load-project-yaml.ipynb [hunk 3]
```diff
@@ -0,0 +1,1359 @@
+{
```
Comment (GiladShapira94) [inline docs/projects/load-project-yaml.ipynb:1]: <ol><li><a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#load-or-create-a-new-project-based-on-an-existing-project-yaml-file" target="_blank">Load </a>a project from remote URL</li><li><a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#get-the-function-object" target="_blank">Get the function object</a> - Get a function object</li><li><a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#run-the-function" target="_blank">Run the function</a> - Run project functions</li><li><a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#build-the-function" target="_blank">Build the function</a> - Build project functions</li><li><a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#deploy-the-function" target="_blank">Deploy the function</a> - Deploy project functions</li><li><a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#run-the-workflow" target="_blank">Run the workflow</a> - Run project workflow</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (GiladShapira94) [inline docs/projects/load-project-yaml.ipynb:1]: <p>need to add remotely to <a href="https://app.reviewnb.com/mlrun/mlrun/pull/3640/#running-the-function-locally" target="_blank">run jobs locally</a>  &nbsp;and remotely</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlrun/mlrun/pull/3640/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (jillnogold) [reply docs/projects/load-project-yaml.ipynb:1]: <p>fixed</p>

File: docs/projects/automate-project-git-source.ipynb (hunk 1 above)
Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: <p>updated, pls see</p>

Comment (jillnogold) [reply docs/projects/automate-project-git-source.ipynb:1]: <p>fixed</p>

File: docs/projects/load-project-yaml.ipynb (hunk 3 above)
Comment (jillnogold) [reply docs/projects/load-project-yaml.ipynb:1]: <p>added</p>

**Comment by GiladShapira94** [comment]
Approved 💪

**Pull request merged**